contracts/Marketplace.sol

```diff
@@ -256,19 +268,23 @@ contract Marketplace is Collateral, Proofs {
 
   function proofEnd(SlotId slotId) public view returns (uint256) {
     Slot memory slot = _slot(slotId);
-    uint256 end = _end(_toEndId(slot.requestId));
+    uint256 end = requestEnd(slot.requestId);
     if (_slotAcceptsProofs(slotId)) {
       return end;
     } else {
       return Math.min(end, block.timestamp - 1);
     }
```

Comment:
I believe we need to check `_requestAcceptsProofs` as this handles edge cases, eg when a request is cancelled but the state hasn't been updated. The only time proof end is set otherwise is when a request is failed (via `freeSlot`).

Reply:
Nice catch!